run_bmi_unit_test.py

```diff
@@ -416,4 +416,4 @@ def bmi_except(fstring):
 print ("\n Total BMI function PASS: " + str(pass_count))
 print (" Total BMI function FAIL: " + str(fail_count))
 for ff in fail_list:
-    print ("  " + ff)   
```

Comment:
A couple unexpected outcomes:

1. `set_value` is run before `get_value`, making it impossible to tell if `get_value` returns sensical values for each var
2. `update_until` is given a value of 100, but it's unclear whether this is 100 time steps or 100 seconds. The output is ambiguous:
```
 updating...        time 3600.0
 updating until...  time 100.0
```

Reply:
1. Not sure that I'm following here...`get_value()` would just then return `NULL` or some initial value set in `initalize()`, etc.
2. I have updated both py scripts to more specific about time info prints

Reply:
@madMatchstick that is a good point re: 1. Can we run the `get/set` tests after the `update()` test? That way we could evaluate all of the var values?

Reply:
@SnowHydrology re re 1: The outcome would be the same. Just to be clear, we are running `set/get` for ALL var roles here, regardless of input vs out,
```
for var_name in (bmi.get_output_var_names() + bmi.get_input_var_names()):     
    print (" " + var_name + ":" )

    #-------------------------------------------------------------------
    # set_value()
    try:
        this_set_value = -99.0
        bmi.set_value(var_name, this_set_value)
        print ("  set value: " + str(this_set_value))

        if var_name_counter == 0: 
            pass_count += 1
    except:
        bmi_except('set_value()')

    #-------------------------------------------------------------------
    # get_value()
    try:
        that_get_value = bmi.get_value(var_name)
        # check if set_value() passed then see if get/set values match
        if 'set_value()' not in fail_list:
            if that_get_value == this_set_value:
                print ("  get value: " + str(that_get_value) + " (values match)")
            else: 
                print ("  get value: " + str(that_get_value) + " (values DO NOT match)")
        else:
            print ("  get value: " + str(that_get_value))      
        if var_name_counter == 0: 
            pass_count += 1
    except:
        bmi_except('get_value()')
```
Perhaps the approach you're after is something more like,
1. `set` some values for input vars
2. run some `update` calls
3. `get` values for outputs

which is typically how bmi control functions are arranged, yes.  But, as far as "unit testing" goes, figured we are just focused on behavior of function defs themselves. 

Besides, I wouldn't be able to keep all the `bmi.control()` functions together under 1 nice neat printout section :),
```
#-------------------------------------------------------------------
#-------------------------------------------------------------------
# BMI: Control Functions
#-------------------------------------------------------------------
#-------------------------------------------------------------------   
print ("\nCONTROL FUNCTIONS\n*****************")    
```

Anyways, let me know if I'm misunderstanding what you are asking...

Reply:
Shelving test updates for future PR